daemon/src/engine/sys/sdl_input.cpp

```diff
@@ -569,94 +567,63 @@ static keyNum_t IN_TranslateSDLToQ3Key( SDL_Keysym *keysym, bool down )
 }
 
 /*
-===============
-IN_GobbleMotionEvents
-===============
-*/
-static void IN_GobbleMotionEvents()
-{
-	SDL_Event dummy[ 1 ];
-
-	// Gobble any mouse motion events
-	SDL_PumpEvents();
-	while ( SDL_PeepEvents( dummy, 1, SDL_GETEVENT,
-	                        SDL_MOUSEMOTION, SDL_MOUSEMOTION ) ) { }
-}
+ * When this is true, the cursor is free to move and cgame will receive
+ * mouse coordinates.
+ * When this is false, the cursor will be locked in the window and cgame
+ * will receive deltas
+ */
+static bool cursor_active = true;
 
 /*
-===============
-IN_ActivateMouse
-===============
-*/
-static void IN_ActivateMouse()
+ * Activates the cursor, the system cursor will still be disabled and
+ * the game is responsible of drawing the cursor image
+ */
+void IN_SetCursorActive( bool active )
 {
 	if ( !mouseAvailable || !SDL_WasInit( SDL_INIT_VIDEO ) )
 	{
 		return;
 	}
 
-	if ( !mouseActive )
-	{
-		SDL_ShowCursor( 0 );
-
-		SDL_SetRelativeMouseMode( SDL_TRUE );
-		SDL_SetWindowGrab( window, SDL_TRUE );
-
-		IN_GobbleMotionEvents();
-	}
-
-	// in_nograb makes no sense in fullscreen mode
-	if ( !cls.glconfig.isFullscreen )
-	{
-		if ( in_nograb->modified || !mouseActive )
-		{
-			if ( in_nograb->integer )
-			{
-				SDL_SetWindowGrab( window, SDL_FALSE );
-			}
-			else
-			{
-				SDL_SetWindowGrab( window, SDL_TRUE );
-			}
+	SDL_ShowCursor( SDL_DISABLE );
+	auto grab = active || in_nograb->integer ? SDL_FALSE : SDL_TRUE;
+	SDL_SetRelativeMouseMode( grab );
+	SDL_SetWindowGrab( window, grab );
 
-			in_nograb->modified = false;
-		}
-	}
+	cursor_active = active;
+}
 
-	mouseActive = true;
+/*
+ * Moves the mouse at the center of the window
+ */
+void IN_CenterMouse()
+{
+	int w, h;
+	SDL_GetWindowSize( window, &w, &h );
+	SDL_WarpMouseInWindow( window, w / 2, h / 2 );
 }
 
 /*
-===============
-IN_DeactivateMouse
-===============
-*/
-void IN_DeactivateMouse( bool showCursor )
+ * Toggles forced cursor, this ensure that the cursor is active and visible
+ * even when not in cursor mode
+ */
+static void ForceCursor(bool force)
```

Comment:
I don't think it would be a problem, since the window will most likely not have focus.
TBH I think it would be cleaner to switch cursor mode just once when needed than having all those checks at every frame, but that's how the old code did it and didn't know how to change this.